presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -331,31 +331,31 @@ public static HiveColumnStatistics fromMetastoreApiColumnStatistics(ColumnStatis
             LongColumnStatsData longStatsData = columnStatistics.getStatsData().getLongStats();
             OptionalLong min = longStatsData.isSetLowValue() ? OptionalLong.of(longStatsData.getLowValue()) : OptionalLong.empty();
             OptionalLong max = longStatsData.isSetHighValue() ? OptionalLong.of(longStatsData.getHighValue()) : OptionalLong.empty();
-            OptionalLong nullsCount = longStatsData.isSetNumNulls() ? OptionalLong.of(longStatsData.getNumNulls()) : OptionalLong.empty();
+            OptionalLong nullsCount = longStatsData.isSetNumNulls() && longStatsData.getNumNulls() >= 0l ? OptionalLong.of(longStatsData.getNumNulls()) : OptionalLong.empty();
```

Comment:
- please add a feature toggle to `HiveClientConfig` like `hive.negative-count-statistics-enabled`, so when this feature toggle is not set `-1` will mean that stats are corrupted
- please compare exactly against `-1` here

Reply:
@kokosing , thanks.
So, from my understanding: 
When `hive.negative-count-statistics-enabled = true`, it means that when we met -1, we didn't corrupt the query, just ignore it.
When `hive.negative-count-statistics-enabled = false`, we should consider this -1 as corrupted and thus failing my query(if `hive.ignore-corrupted-statistics=false`)

Did I get your point ?

Reply:
@kokosing  , i have resubmit my code. I think that the name `hive.negative-count-statistics-enabled` is not value accurate, because it is just used indicate a single negative value -1, instead of all negative values.
what's your opinion?

Reply:
We talked offline with @kokosing. Let's not add a new toggle for that.